lib/splitclient-rb/engine/matchers/semver.rb

```diff
@@ -0,0 +1,171 @@
+# frozen_string_literal: true
+
+module SplitIoClient
+  class Semver
+    METADATA_DELIMITER = '+'
+    PRE_RELEASE_DELIMITER = '-'
+    VALUE_DELIMITER = '.'
+
+    attr_reader :major, :minor, :patch, :pre_release, :is_stable, :old_version
+
+    #
+    # Class Initializer
+    #
+    # @param version [String] raw version as read from splitChanges response.
+    def initialize(version)
+      @major = 0
+      @minor = 0
+      @patch = 0
+      @pre_release = []
+      @is_stable = false
+      @old_version = version
+      parse
```

Comment:
please don't parse at constructor level.
if you have a `RuntimeError` at this level you are not catching it.

Reply:
will do